oneOf

Chadtech opened the pull request:
Heres a function I used in a project at work recently, I thought could be a good addition to this repo. it takes a list of `Maybe` and returns the first `Just` if any of them are `Just`.

I think we've used it in a few places, but one of them is during routing. We have two methods of parsing a url. If either of them succeed, they will return a `Just Route`. We want to succeed if either of them work, and we arent picky about which one. 

I cant find the code now, but I think we also wanted to use it in a case where we want to provide a summary of some user data, which has many optional fields. So if any of the data exists, we display just the priority one, if none of the data exists, we display something different entirely.

Also, my elm-format made lots of little formatting changes to this repo. Sorry.

agrafix commented:
Thanks for the PR. Although you noted that `elm-format` made lot's of unrelated changes, it would be helpful to split those changes into a different PR.

The function can be implemented a lot simpler using existing combinators:

```elm
oneOf : List (Maybe a) -> Maybe a
oneOf = List.foldr (Maybe.Extra.or) Nothing
```

I'm also not sure if `oneOf` is the correct name, in Haskell it's usually called `msum` or `asum`, but that does not need to guide this name. Still, `oneOf` doesn't really reflect what the function does. How about `firstJust`?

Chadtech commented:
Thanks @agrafix ! I changed the implementation and name as you suggested.

Regarding the elm-format changes, how do I split the changes into a different PR? Is it a really problem if the formatting is changed by this PR?
